### PR TITLE
Add CRUD support for spaces groups

### PR DESCRIPTION
**Pull request by RedbackThomson**
### Description of your changes

Rebased on top of https://github.com/upbound/up/pull/457

Add support for `up group create/list/get/delete` using the selected spaces kubeconfig.

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

```bash
$ go run ./cmd/up group list
NAME   PROTECTED

$ go run ./cmd/up group create nick-unprotected
nick-unprotected created
$ go run ./cmd/up group create nick-protected --deletion-protection
nick-protected created
$ go run ./cmd/up group list
NAME               PROTECTED
nick-protected     true
nick-unprotected   false

$ go run ./cmd/up group get nick-protected
NAME             PROTECTED
nick-protected   true

$ go run ./cmd/up group delete nick-protected
up: error: group.deleteCmd.Run(): Deletion protection is enabled on the specified group. Use '--force' to delete anyway.
exit status 1
$ go run ./cmd/up group delete nick-protected --force
nick-protected deleted
$ go run ./cmd/up group delete nick-unprotected
nick-unprotected deleted
```

### CODE REVIEW

File: cmd/up/group/cmd.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+// Cmd contains commands for interacting with groups.
+type Cmd struct {
+	Create createCmd `cmd:"" help:"Create a group."`
+	Delete deleteCmd `cmd:"" help:"Delete a group."`
+	List   listCmd   `cmd:"" help:"List groups in the space."`
+	Get    getCmd    `cmd:"" help:"Get a group."`
+
+	Short       bool   `short:"s" env:"UP_SHORT" name:"short" help:"Short output."`
+	KubeContext string `env:"UP_CONTEXT" default:"upbound" name:"context" help:"Kubernetes context to operate on."`
```
Comment (sttts) [inline cmd/up/group/cmd.go:68]: is this one used?

I think we have to uniformely wire this through into `GetCurrentContext(ctx)`.

Comment (sttts) [reply cmd/up/group/cmd.go:68]: Maybe also an idea, as we get more and more CRUD commands, we could create some common set of flags for CRUD.

Comment (RedbackThomson) [reply cmd/up/group/cmd.go:68]: Actually neither of these other CLI args are used - I was just looking over that myself. What is the purpose of this arg, anyway? Just another way to override the current context of the kubeconfig?

File: cmd/up/group/delete.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+// Run executes the create command.
+func (c *deleteCmd) Run(ctx context.Context, printer upterm.ObjectPrinter, upCtx *upbound.Context, p pterm.TextPrinter) error { // nolint:gocyclo
+	// get context
+	_, currentProfile, _, err := upCtx.Cfg.GetCurrentContext(ctx)
+	if err != nil {
+		return err
+	}
+	if currentProfile == nil {
+		return errors.New(profile.NoSpacesContextMsg)
+	}
```
Comment (sttts) [inline cmd/up/group/delete.go:50]: these need more checks, namely that we are not on a controlplane, but only on a group.

This is actually up for debate. Current theory is that we want upper check. But of course it forces you into more context checks. You wouldn't be able to list groups while being on a controlplane.

Comment (RedbackThomson) [reply cmd/up/group/delete.go:50]: Cool. I had that originally but it felt weirdly limiting to not allow that. There is no technical reason why we couldn't, ofc. But I won't add more heat into that debate for now lol

**Review by sttts — APPROVED** [review]

**Pull request merged**